tests/test_federated_aws_cli.py

```diff
@@ -3,57 +3,76 @@
+    help_result = runner.invoke(cli.main, ["--help"])
     assert help_result.exit_code == 0
-    assert 'Show this message and exit.' in help_result.output
+    assert "Show this message and exit." in help_result.output
 
 
 def test_parse_config():
-    good_arn = 'arn:aws:iam::123456789012:role/MyRole'
-    bad_arn = 'bogus'
-    good_config_content = '''[DEFAULT]
+    good_config_content = """[DEFAULT]
 well_known_url = http://auth.example.com/.well-known/openid-configuration
 client_id = abcdefghijklmnopqrstuvwxyz012345
-idtoken_for_roles_url = https://example.com/roles'''
+idtoken_for_roles_url = https://example.com/roles"""
+
+    bad_arn = "bogus"
+    bad_output_setting = good_config_content + "\noutput = blahblahblah"
+    good_arn = "arn:aws:iam::123456789012:role/MyRole"
+    missing_config_setting = """[DEFAULT]
+well_known_url = http://auth.example.com/.well-known/openid-configuration
+    """
+
     cases = {
-        'malformed ini': {
-            'filename': os.path.expanduser(os.path.join("~", ".malformed")),
-            'content': '- a z=x:\ny: "${z}"',
-            'args': ['-c', os.path.expanduser(os.path.join("~", ".malformed")), '-r', good_arn]
+        "malformed ini": {
+            "content": "- a z=x:\ny: \"${z}\"",
+            "args": ["-r", good_arn]
+        },
+        "missing config setting": {
+            "content": missing_config_setting,
+            "args": ["-r", good_arn]
         },
-        'missing config setting': {
-            'filename': os.path.expanduser(os.path.join("~", ".missing_config_setting")),
-            'content': '[DEFAULT]\nwell_known_url = http://auth.example.com/.well-known/openid-configuration\n',
-            'args': ['-c', os.path.expanduser(os.path.join("~", ".missing_config_setting")), '-r', good_arn]
+        "bad role arn": {
+            "content": good_config_content,
+            "args": ["-r", bad_arn]
+        },
+        "bad output setting": {
+            "content": bad_output_setting,
+            "args": ["-r", good_arn]
         },
-        'bad role arn': {
-            'filename': os.path.expanduser(os.path.join("~", ".good_config")),
-            'content': good_config_content,
-            'args': ['-c', os.path.expanduser(os.path.join("~", ".good_config")), '-r', bad_arn]
-        }
     }
 
     runner = CliRunner()
     with runner.isolated_filesystem():
         results = {}
         for case in cases:
-            with open(cases[case]['filename'], 'w') as f:
-                f.write(cases[case]['content'])
-            results[case] = runner.invoke(cli.main, cases[case]['args'])
+            args = cases[case]["args"]
+
+            # Use a random filename, if "-c" isn"t specified in the arguments above
+            filename = str(uuid.uuid4())
+            if "-c" not in args:
+                args += ["-c", filename]
+
+            with open(filename, "w") as f:
+                f.write(cases[case]["content"])
```

Comment:
Does anything cleanup these UUID files? If these are meant to be temporary files (since they have random names) we should use [`tempfile`](https://docs.python.org/3.8/library/tempfile.html) and clean them up.

Reply:
Yes, `with runner.isolated_filesystem()` creates a temporary filesystem that cleans up after itself.